CI: TestGetIdentity/Duplicated_identity failure

### Description

Failure link https://app.travis-ci.com/github/cilium/cilium/jobs/594047847, this issue is replicated locally as well.

```
=== RUN   TestGetIdentity/Duplicated_identity
    identity_test.go:218: Expected key 10, got 11
    identity_test.go:218: Expected key 10, got 11
    --- FAIL: TestGetIdentity/Duplicated_identity (0.03s)
```

From the first look, the test expectation is to get key 10, however, the underlying `get` method didn't guarantee any particular order while retrieving element from map.

https://github.com/cilium/cilium/blob/master/pkg/k8s/identitybackend/identity.go#L219-L242

--- a/pkg/k8s/identitybackend/identity.go
+++ b/pkg/k8s/identitybackend/identity.go
@@ -7,6 +7,7 @@ import (
 	"context"
 	"fmt"
 	"reflect"
+	"sort"
 	"strconv"
 	"strings"
 
@@ -216,8 +217,9 @@ func (c *crdLock) Comparator() interface{} {
 	return nil
 }
 
-// get returns the first identity found for the given set of labels as we might
-// have duplicated entries identities for the same set of labels.
+// get returns the identity found for the given set of labels.
+// In the case of duplicate entries, return an identity entry
+// from a sorted list.
 func (c *crdBackend) get(ctx context.Context, key allocator.AllocatorKey) *v2.CiliumIdentity {
 	if c.Store == nil {
 		return nil
@@ -227,6 +229,20 @@ func (c *crdBackend) get(ctx context.Context, key allocator.AllocatorKey) *v2.Ci
 	if err != nil || len(identities) == 0 {
 		return nil
 	}
+
+	sort.Slice(identities, func(i, j int) bool {
+		left, ok := identities[i].(*v2.CiliumIdentity)
+		if !ok {
+			return false
+		}
+
+		right, ok := identities[j].(*v2.CiliumIdentity)
+		if !ok {
+			return false
+		}
+
+		return left.CreationTimestamp.Before(&right.CreationTimestamp)
+	})
 
 	for _, identityObject := range identities {
 		identity, ok := identityObject.(*v2.CiliumIdentity)

--- a/pkg/k8s/identitybackend/identity_test.go
+++ b/pkg/k8s/identitybackend/identity_test.go
@@ -110,6 +110,9 @@ func createCiliumIdentity(id int, labels map[string]string) v2.CiliumIdentity {
 	return v2.CiliumIdentity{
 		ObjectMeta: v1.ObjectMeta{
 			Name: strconv.Itoa(id),
+			CreationTimestamp: v1.Time{
+				Time: time.Now(),
+			},
 		},
 		SecurityLabels: getLabelsMap(labels),
 	}

